### PR TITLE
[FIX] web_editor: revert text color previews

### PR DESCRIPTION
**Pull request by bso-odoo**
Before this commit when a text color was applied on a text within
which an already colored text already existed, the inner coloring was
lost during preview. All text was therefore set to the initial coloring
of the external selected text.

After this commit the "reset" of the text color preview is done by
undoing the draft "in progress" history step that contains the preview
changes, but that is not stored within the history itself.

Steps to reproduce:
- drop the Text snippet
- select a word in the middle of a sentence (e.g. "personality")
- set text color to red
- select several words containing the colored one (e.g. "story with
personality for potential")
- open text color palette
- hover over colors but do not click
- leave color palette
=> colored word is not colored anymore

Steps to reproduce a secondary problem also fixed by this commit:
- drop the Text snippet
- select text surrounding some bold (e.g. "have a personality. Consider")
- open text color palette
- quickly hover through several colors (i.e. move pointer downwards)
=> text selection range changed between some of the color previews

task-2742112 (was task-2666200)

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by bso-odoo** [comment]
@Goaman I have been told you are the right person to review this PR.

File: addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js [hunk 1]
```diff
@@ -733,6 +733,20 @@ export class OdooEditor extends EventTarget {
         step.mutations = step.mutations.slice(0, until);
         this._toRollback = false;
     }
+    /**
+     * Undo the current non-recorded draft step.
+     */
+    historyRevertDraft() {
```
Comment (Goaman) [inline addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js:739]: I would call it `historyRevertCurrentStep`.

**Comment by Goaman** [comment]
Other than the name of the function, the fix looks great. Nice job :+1:

File: addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js (hunk 1 above)
Comment (bso-odoo) [reply addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js]: @Goaman Suggested change done

**Comment by bso-odoo** [comment]
@Goaman I just rebased this. Can it be r+'ed ?

**Comment by bso-odoo** [comment]
@Goaman I just re-rebased this. Can it be r+'ed ?

**Comment by Goaman** [comment]
Fine by me. cc @dmo-odoo r+ ?

**Comment by dmo-odoo** [comment]
@robodoo r+

**Pull request closed without merging**